PR build status:
```cpp
src/server/scripts/Spells/spell_mage.cpp:142:36: fatal error: 
      unused parameter 'spellInfo' [-Wunused-parameter]
    bool Validate(SpellInfo const* spellInfo) override
                                   ^
```
TC SQL codestyle; it is not necessary to comment the difference between the 'delete' part and the 'insert' part. Use only a first line comment to say something about the whole SQL content.

Example:
```sql
-- mage Alter Time spell
DELETE FROM `spell_script_names` WHERE `ScriptName` IN ('spell_mage_alter_time_aura', 'spell_mage_alter_time_active');
INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
(110909,'spell_mage_alter_time_aura'),
(127140,'spell_mage_alter_time_active');
```

--- a/sql/updates/world/master/2021_05_27_00_world_mage_spell_alter_time.sql
+++ b/sql/updates/world/master/2021_05_27_00_world_mage_spell_alter_time.sql
@@ -1,0 +1,7 @@
+-- mage Alter Time spell
+DELETE FROM `spell_script_names` WHERE `ScriptName` IN ('spell_mage_alter_time_aura', 'spell_mage_alter_time_active');
+INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
+(110909,'spell_mage_alter_time_aura'),
+(127140,'spell_mage_alter_time_active'),
+(342246,'spell_mage_alter_time_aura'),
+(342247,'spell_mage_alter_time_active');

--- a/src/common/Logging/Log.cpp
+++ b/src/common/Logging/Log.cpp
@@ -56,7 +56,7 @@ Appender* Log::GetAppenderByName(std::string const& name)
     return it == appenders.end() ? nullptr : it->second.get();
 }
 
-void Log::CreateAppenderFromConfig(std::string const& appenderName)
+void Log::CreateAppenderFromConfigLine(std::string const& appenderName, std::string const& options)
 {
     if (appenderName.empty())
         return;
@@ -64,7 +64,6 @@ void Log::CreateAppenderFromConfig(std::string const& appenderName)
     // Format = type, level, flags, optional1, optional2
     // if type = File. optional1 = file and option2 = mode
     // if type = Console. optional1 = Color
-    std::string options = sConfigMgr->GetStringDefault(appenderName, "");
 
     Tokenizer tokens(options, ',');
     auto iter = tokens.begin();
@@ -109,16 +108,20 @@ void Log::CreateAppenderFromConfig(std::string const& appenderName)
     }
 }
 
-void Log::CreateLoggerFromConfig(std::string const& appenderName)
+void Log::CreateAppenderFromConfig(std::string const& appenderName)
 {
-    if (appenderName.empty())
+    CreateAppenderFromConfigLine(appenderName, sConfigMgr->GetStringDefault(appenderName, ""));
+}
+
+void Log::CreateLoggerFromConfigLine(std::string const& loggerName, std::string const& options)
+{
+    if (loggerName.empty())
         return;
 
     LogLevel level = LOG_LEVEL_DISABLED;
     uint8 type = uint8(-1);
 
-    std::string options = sConfigMgr->GetStringDefault(appenderName, "");
-    std::string name = appenderName.substr(7);
+    std::string name = loggerName.substr(7);
 
     if (options.empty())
     {
@@ -170,6 +173,11 @@ void Log::CreateLoggerFromConfig(std::string const& appenderName)
             fprintf(stderr, "Error while configuring Appender %s in Logger %s. Appender does not exist", str.c_str(), name.c_str());
         ss >> str;
     }
+}
+
+void Log::CreateLoggerFromConfig(std::string const& loggerName)
+{
+    CreateLoggerFromConfigLine(loggerName, sConfigMgr->GetStringDefault(loggerName, ""));
 }
 
 void Log::ReadAppendersFromConfig()

--- a/src/common/Logging/Log.h
+++ b/src/common/Logging/Log.h
@@ -97,6 +97,9 @@ class TC_COMMON_API Log
         std::string const& GetLogsDir() const { return m_logsDir; }
         std::string const& GetLogsTimestamp() const { return m_logsTimestamp; }
 
+        void CreateAppenderFromConfigLine(std::string const& name, std::string const& options);
+        void CreateLoggerFromConfigLine(std::string const& name, std::string const& options);
+
     private:
         static std::string GetTimestampStr();
         void write(std::unique_ptr<LogMessage>&& msg) const;

--- a/src/server/scripts/Spells/spell_mage.cpp
+++ b/src/server/scripts/Spells/spell_mage.cpp
@@ -33,11 +33,15 @@
 
 enum MageSpells
 {
+    SPELL_MAGE_ALTER_TIME_AURA                   = 110909,
+    SPELL_MAGE_ALTER_TIME_VISUAL                 = 347402,
+    SPELL_MAGE_ARCANE_ALTER_TIME_AURA            = 342246,
     SPELL_MAGE_ARCANE_BARRAGE_ENERGIZE           = 321529,
     SPELL_MAGE_ARCANE_BARRAGE_R3                 = 321526,
     SPELL_MAGE_ARCANE_CHARGE                     = 36032,
     SPELL_MAGE_ARCANE_MAGE                       = 137021,
     SPELL_MAGE_BLAZING_BARRIER_TRIGGER           = 235314,
+    SPELL_MAGE_BLINK                             = 1953,
     SPELL_MAGE_CAUTERIZE_DOT                     = 87023,
     SPELL_MAGE_CAUTERIZED                        = 87024,
     SPELL_MAGE_CONE_OF_COLD                      = 120,
@@ -54,6 +58,7 @@ enum MageSpells
     SPELL_MAGE_LIVING_BOMB_EXPLOSION             = 44461,
     SPELL_MAGE_LIVING_BOMB_PERIODIC              = 217694,
     SPELL_MAGE_MANA_SURGE                        = 37445,
+    SPELL_MAGE_MASTER_OF_TIME                    = 342249,
     SPELL_MAGE_REVERBERATE                       = 281482,
     SPELL_MAGE_RING_OF_FROST_DUMMY               = 91264,
     SPELL_MAGE_RING_OF_FROST_FREEZE              = 82691,
@@ -78,6 +83,85 @@ enum MiscSpells
     SPELL_SHAMAN_EXHAUSTION                      = 57723,
     SPELL_SHAMAN_SATED                           = 57724,
     SPELL_MAGE_CHILLED                           = 205708
+};
+
+// 110909 - Alter Time Aura
+// 342246 - Alter Time Aura
+class spell_mage_alter_time_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_alter_time_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_MAGE_ALTER_TIME_VISUAL,
+            SPELL_MAGE_MASTER_OF_TIME,
+            SPELL_MAGE_BLINK,
+        });
+    }
+
+    void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetTarget();
+        _health = unit->GetHealth();
+        _pos = unit->GetPosition();
+    }
+
+    void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        Unit* unit = GetTarget();
+        if (unit->GetDistance(_pos) <= 100.0f && GetTargetApplication()->GetRemoveMode() == AURA_REMOVE_BY_EXPIRE)
+        {
+            unit->SetHealth(_health);
+            unit->NearTeleportTo(_pos);
+
+            if (unit->HasAura(SPELL_MAGE_MASTER_OF_TIME))
+            {
+                SpellInfo const* blink = sSpellMgr->AssertSpellInfo(SPELL_MAGE_BLINK, DIFFICULTY_NONE);
+                unit->GetSpellHistory()->ResetCharges(blink->ChargeCategoryId);
+            }
+            unit->CastSpell(unit, SPELL_MAGE_ALTER_TIME_VISUAL);
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_mage_alter_time_aura::OnApply, EFFECT_0, SPELL_AURA_OVERRIDE_ACTIONBAR_SPELLS, AURA_EFFECT_HANDLE_REAL);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_mage_alter_time_aura::AfterRemove, EFFECT_0, SPELL_AURA_OVERRIDE_ACTIONBAR_SPELLS, AURA_EFFECT_HANDLE_REAL);
+    }
+
+private:
+    uint64 _health = 0;
+    Position _pos;
+};
+
+// 127140 - Alter Time Active
+// 342247 - Alter Time Active
+class spell_mage_alter_time_active : public SpellScript
+{
+    PrepareSpellScript(spell_mage_alter_time_active);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_MAGE_ALTER_TIME_AURA,
+            SPELL_MAGE_ARCANE_ALTER_TIME_AURA,
+        });
+    }
+
+    void RemoveAlterTimeAura(SpellEffIndex /*effIndex*/)
+    {
+        Unit* unit = GetCaster();
+        unit->RemoveAura(SPELL_MAGE_ALTER_TIME_AURA, ObjectGuid::Empty, 0, AURA_REMOVE_BY_EXPIRE);
+        unit->RemoveAura(SPELL_MAGE_ARCANE_ALTER_TIME_AURA, ObjectGuid::Empty, 0, AURA_REMOVE_BY_EXPIRE);
+    }
+
+    void Register() override
+    {
+        OnEffectHit += SpellEffectFn(spell_mage_alter_time_active::RemoveAlterTimeAura, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
 };
 
 // 44425 - Arcane Barrage
@@ -982,6 +1066,8 @@ class spell_mage_water_elemental_freeze : public SpellScript
 
 void AddSC_mage_spell_scripts()
 {
+    RegisterAuraScript(spell_mage_alter_time_aura);
+    RegisterSpellScript(spell_mage_alter_time_active);
     RegisterSpellScript(spell_mage_arcane_barrage);
     RegisterSpellScript(spell_mage_arcane_charge_clear);
     RegisterSpellScript(spell_mage_arcane_explosion);

